sentry/src/main/java/io/sentry/Dsn.java

```diff
@@ -49,6 +50,7 @@ URI getSentryUri() {
 
   Dsn(@Nullable String dsn) throws IllegalArgumentException {
     try {
+      Objects.requireNonNull(dsn, "The DSN is required.");
```

Comment:
Not sure this makes sense since the param is `@Nullable`.

Reply:
It would fail further down with a NPE if I recall correctly.

Reply:
So yeah maybe this should not be `@Nullable`

Reply:
It'd and there's a try-catch, so we are just doing an early check that has the same result.

Reply:
The idea here was to generate an exception that's easier to understand and pinpoint the problem. Want me to roll back?

Reply:
If it gives a better error message, fine with it, but I'd change the annotation to non nullable.

Reply:
This might cause some more discussions around the implementation. Mind if we move it to a separate task?

Reply:
yes, let's roll back this change then for now, right now its semantically wrong because its nullable but it throws if it is

Reply:
Created: https://github.com/getsentry/sentry-java/issues/2262 to track